### PR TITLE
chore: Add operator termination duration metrics

### PR DESCRIPTION
**Pull request by jigisha620**
*Issue #, if available:*

*Description of changes:*
Adding new metrics for monitoring termination duration for objects. Metrics added -
1. operator_termination_current_time_seconds - The current amount of time in seconds that an object has been in terminating state
2. operator_termination_duration_seconds - The amount of time taken by an object to terminate completely

Added tests for the changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: status/controller_test.go [hunk 1]
```diff
@@ -33,7 +33,21 @@ var _ = Describe("Controller", func() {
 		controller = status.NewController[*TestObject](kubeClient, recorder)
 		ctx = log.IntoContext(context.Background(), ginkgo.GinkgoLogr)
 	})
+	It("should emit termination metrics when deletion timestamp is set", func() {
```
Comment (jonathan-innis) [inline status/controller_test.go:36]: This doesn't test that we are actually emitting the metric, right?

Comment (jigisha620) [reply status/controller_test.go:36]: It does. Added more checks in the test to validate duration seconds in greater than 0.

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**